### PR TITLE
Tests/PayPro tests for created tx formation

### PR DESCRIPTION
**Pull request by bsnowden3**
This PR, tests how a txp is form based on a stubbed flow of PayPro requests.

### CODE REVIEW

**Comment by Gamboster** [comment]
This PR has errors with tslint. Would you like to fix them?

**Comment by bsnowden3** [comment]
> This PR has errors with tslint. Would you like to fix them?

Will fix them now.

File: src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1605 @@
...
+import { OnGoingProcessProvider } from '../../../../providers/on-going-process/on-going-process';
+import { PayproProvider } from '../../../../providers/paypro/paypro';
+import { ReplaceParametersProvider } from '../../../../providers/replace-parameters/replace-parameters';
+import { TxFormatProvider } from '../../../../providers/tx-format/tx-format';
+import { WalletProvider } from '../../../../providers/wallet/wallet';
+import { ConfirmCardPurchasePage } from '../confirm-card-purchase/confirm-card-purchase';
+
+import * as moment from 'moment';
+
+fdescribe('ConfirmCardPurchasePage', () => {
```
Comment (Gamboster) [inline src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.spec.ts]: forgot a `fdescribe` here

Comment (bsnowden3) [reply src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.spec.ts]: fixed

File: src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.ts [hunk 2]
```diff
@@ -633,10 +636,15 @@ export class ConfirmCardPurchasePage extends ConfirmPage {
...
+  public async onWalletSelect(wallet): Promise<void> {
     this.wallet = wallet;
     this.isERCToken = this.currencyProvider.isERCToken(this.wallet.coin);
     const email = await this.promptEmail();
+    if (email) {
+      await this.initialize(wallet, email).catch(() => {});
+    } else {
+      this.setEmail();
+    }
     await this.initialize(wallet, email).catch(() => {});
```
Comment (cmgustavo) [inline src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.ts]: If email exists, `this.initialize` will execute twice.

**Pull request closed without merging**